src/assets/src/hooks/useWebSocket.ts

```diff
@@ -33,16 +35,35 @@ export const useWebSocket = <T>(url: string, onUpdate: (content: T) => void, onD
                     break;
             }
         }
-        ws.onclose = (e: CloseEvent) => {
-            if (e.code === 1000) return;
-            console.error(e);
-            setError(new Error(closeCodes[e.code] ?? e.code.toString()));
+        ws.onclose = (e) => {
+            if (e.code === 4404) { // Equivalent of HTTP 404
+                setError(new Error("The resource you're looking for does not exist. Maybe it was deleted?"));
+                ws.close();
+            } else if (e.code === 1011) { // Equivalent of HTTP 500
+                setError(new Error("Sorry, there was a system error. Please try refreshing. If you continue to receive this error, contact the ITS Service Center."));
+                ws.close();
+            } else if (e.code === 1001) { // Page refreshed (Firefox)
+                ws.close();  
```

Comment:
Oh, was this where we saw it was a different code on `localhost` compared to when deployed? If that's the case, I guess we can leave this as is?

Reply:
Yes, it's 1000 if DEBUG=True and 1001 if DEBUG=False.

Reply:
Hm, little confusing to not handle `DEBUG=True` the same way, but I guess handling `1000` isn't easy either...

Reply:
What do you think, could we still handle `1000` too here? Seems like because you're setting `onclose` to `null` in the cleanup, we wouldn't trigger the same error messages when the page is left?

Reply:
We could do something like this, and I tried it on a couple desktop browsers. I guess I'd just be concerned about unintended consequences:

```
            } else if ([1000, 1001].includes(e.code)) { // Page refreshed (Firefox)
                ws.close();
```

Reply:
It's probably reasonable to handle `1000` the same way - since it indicates a "normal" (non-error) closure, why retry for a non-error? But OTOH, it does mean the connection is closed, which we might want to indicate in some way, and it's possible that a bug in the consumer code could close things out with `1000` unexpectedly.

Reply:
I think I'll handle the `1000` errors as `1001` for now, and take this as an indication that this piece might need some refactoring down the line.

Reply:
In WebKit, when you return to a cached page, it will give error code 1000, which needs to be retried. So I can't handle it the same as 1001 without branching on user agent, which seems overly finnicky and error-prone.

Reply:
I'm a bit confused (sorry), wouldn't both `1000` and `1001` lead to reconnection attempts in this scenario?

Reply:
(Yeah, I would avoid branching on user agent.)

Reply:
1001 doesn't lead to reconnection attempts, it causes `ws.close()` which doesn't incite a reconnect. The idea is that 1001 indicates "Going Away" so there is no need to reconnect. As it stands, 1000 would fall under "Retry all others", and we need it to be retried to work around WebKit's caching (back button) issue.

Reply:
Glad I asked, sorry, I was definitely confused. Making much more sense now.

Reply:
Okay, apologies if this is getting annoying, I'm generally OK with what we have currently, for the record.

That said, what if we moved the `1000` and `1001` handling under "Retry all others", like so:

```
            } else { // Retry all others
                console.error(e);
                if (![1000, 1001].includes(e.code)) {
                    setError(new Error(`The connection unexpectedly closed (${e.code.toString()}). Trying to reconnect...`));    
                }
            }
```

If 1000 and 1001 are expected to happen occasionally on refresh, navigation, and it's safe to try reconnecting, do we need to report them as errors?

Reply:
We don't want to retry 1001, so it won't fall under that branch, but as for 1000: I could choose to retry while suppressing the error, which makes perfect sense on the surface, but when WebKit closes with 1000 upon pressing the "back" button, it ALSO emits an error event with no error data, which would still be displayed on screen. I could choose to suppress the error event message if the error event has no data, but how can we be totally sure that doesn't happen (error event with no error data) in any other circumstances that we would want to know about?

Reply:
Yeah, this is confusing with all the different codes and browser behavior, and it seems like you've thought through the different circumstances. While it bugs me that what we see when working locally doesn't reflect what happens in production, I don't think it's a huge deal, and there don't seem to be easy answers.